src/types/array.ts

```diff
@@ -29,10 +29,12 @@ class SchemaTypeArray<I, T extends SchemaType<I>> extends SchemaType<I[]> {
   /**
    * Casts an array and its child elements.
    *
-   * @param {*} value
+   * @param {*} value_
    * @param {Object} data
    * @return {Array}
    */
+  cast(value_: Exclude<unknown, null | undefined>, data?: unknown): I[];
+  cast(value_?: unknown, data?: unknown): I[] | null | undefined;
   cast(value_?: unknown, data?: unknown): I[] | null | undefined {
     value_ = super.cast(value_, data);
     if (value_ == null) return value_ as null | undefined;
```

Comment:
IMHO the type of `SchemaType#cast` should also be improved to avoid the `as` here.